Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1

```diff
@@ -386,6 +386,174 @@ function Publish-TerminologyDacpac() {
+    $identityId = $identityQueryResult.Table.IdentityID
+
+    if ($identityId) {
+        Write-DosMessage -Level "Information" -Message  "IdentityID for $user`: $identityId"
+    }
+    else {
+        Write-DosMessage -Level "Information" -Message   "User is not in IdentityBASE... Adding user: $user"
+        Invoke-Sql -sqlCommand "INSERT INTO [$metadataDatabase].[CatalystAdmin].[IdentityBASE] (IdentityNM) VALUES (UPPER('$user'))";
+        $identityBaseQuery = "SELECT TOP (1) [IdentityID] FROM [$metadataDatabase].[CatalystAdmin].[IdentityBASE] WHERE UPPER(IdentityNM) = UPPER('$user')"
+        $identityQueryResult = Invoke-SqlCommand -SqlServerAddress $sqlAddress -DatabaseName $metadataDatabase -Query $identityBaseQuery
+        $identityId = $identityQueryResult.Table.IdentityID
+    }
+
+    $dataProcessingServiceRoleId = Get-RoleId -role "DataProcessingServiceUser"
+
+    # POST Terminology data marts to MDS
+    Write-DosMessage -Level "Information" -Message   "Creating Terminology metadata. This could take upwards of ten minutes depending on the availability and capacity of the server. In most cases it takes between 3 and 6 minutes."
+    $terminologyDataMartId = Post-ToMds -discoveryServiceUrl $discoveryServiceUrl -name "Terminology" -path ".\Terminology.json"
+
+    Write-DosMessage -Level "Information" -Message   "Creating SharedTerminology metadata. This could also take upwards of ten minutes, depending on the server. In most cases, it takes between 4-7 minutes."
+    $sharedTerminologyDataMartId = Post-ToMds -discoveryServiceUrl $discoveryServiceUrl -name "SharedTerminology" -path ".\SharedTerminologyFiveNPEs.json"
+
+
+    # POST executions 
+    Write-DosMessage -Level "Information" -Message   "Creating physical tables for Terminology and SharedTerminology data marts"
+    $terminologyBatchExecutionId = Post-ToDps -discoveryServiceUrl $discoveryServiceUrl -dataMartId $terminologyDataMartId
+    $sharedTerminologyBatchExecutionId = Post-ToDps -discoveryServiceUrl $discoveryServiceUrl -dataMartId $sharedTerminologyDataMartId
+
+    # Poll batch executions for 5 minutes to determine if they've been successful
+    $wasSuccessful = Poll-BatchExecutions -terminologyBatchExecutionId $terminologyBatchExecutionId -sharedTerminologyBatchExecutionId $sharedTerminologyBatchExecutionId
+
+    # if DPS role was added for user, remove the role
+    if ($dataProcessingRoleAdded) {
+        Write-DosMessage -Level "Information" -Message   "Removing $roleName role from user $user"
+        $identityRoleBaseQuery = "DELETE FROM [$metadataDatabase].[CatalystAdmin].[IdentityRoleBASE] WHERE IdentityId = $identityId AND RoleID = $dataProcessingServiceRoleId"
+        $identityRoleQueryResult = Invoke-SqlCommand -SqlServerAddress $sqlAddress -DatabaseName $metadataDatabase -Query $identityRoleBaseQuery
+        Write-DosMessage -Level "Information" -Message   "Role removed"
+    }
+
+    if(!$wasSuccessful) {
```

Comment:
It would be nice to have a final success message

Reply:
I thought a final success message would make more sense after the validation completed. I thought it would be confusing to have "Success!" and then have it fail.

Reply:
I was just talking about the mds/dps steps. Not the overall installation